src/io/iohk/scalanet/test/NetUtils.scala

```diff
@@ -0,0 +1,52 @@
+package io.iohk.scalanet
+
+import java.net._
+import java.nio.ByteBuffer
+
+object NetUtils {
+
+  def aRandomAddress(): InetSocketAddress = {
+    val s = new ServerSocket(0)
+    try {
+      new InetSocketAddress("localhost", s.getLocalPort)
+    } finally {
+      s.close()
+    }
+  }
+
+  def isListening(address: InetSocketAddress): Boolean = {
+    try {
+      new Socket(address.getHostName, address.getPort).close()
+      true
```

Comment:
If I understand correctly, the address is used (or listening) if we aren't able to create the socket, which means that we must return false.

Reply:
This is client socket connection to the server.If it succeeds then server is listening